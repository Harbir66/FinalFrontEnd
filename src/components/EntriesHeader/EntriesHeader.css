.main-container {
  width: 60vw;
  height: 10vh;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
}

.column-names {
  width: 60vw;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0;
}

.item {
  width: 6vw;
  margin: 0;
}

.buttons {
  width: 10vw;
  margin: 0 0 0 1vw;
}
