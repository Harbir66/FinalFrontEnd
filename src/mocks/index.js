export const mockCollection = [
  {
    id: 1,
    name: 'collection1',
    fields: 3,
    values: 3,
    allFields: ['field1', 'field2', 'field3'],
  },
  {
    id: 2,
    name: 'collection2',
    fields: 3,
    values: 3,
    allFields: ['field1', 'field2', 'field3'],
  },
];

export const mockHeading = 'Heading';

export const mockFields = ['field1', 'field2', 'field3'];
