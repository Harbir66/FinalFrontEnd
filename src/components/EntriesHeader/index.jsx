import React from 'react';
import './EntriesHeader.css';
import PropTypes from 'prop-types';

function EntriesHeader({ fields }) {
  const firstFourFields = fields.slice(0, 4);
  firstFourFields.unshift('id');
  return (
    <div className="main-container">
      <div className="column-names">
        {firstFourFields.map((field) => (
          <div className="item" key={field}>
            {field}
          </div>
        ))}
      </div>
      <div className="buttons">Actions</div>
    </div>
  );
}

export default EntriesHeader;

EntriesHeader.propTypes = {
  fields: PropTypes.arrayOf(PropTypes.string).isRequired,
};
