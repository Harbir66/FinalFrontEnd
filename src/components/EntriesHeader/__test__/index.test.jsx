import React from 'react';
import { render } from '@testing-library/react';
import EntriesHeader from '..';
import { mockFields } from '../../../mocks';

describe('EntriesHeader', () => {
  it('should render', () => {
    const { asFragment } = render(<EntriesHeader fields={mockFields} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
