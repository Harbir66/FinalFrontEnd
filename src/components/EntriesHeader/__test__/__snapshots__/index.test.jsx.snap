// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntriesHeader should render 1`] = `
<DocumentFragment>
  <div
    class="main-container"
  >
    <div
      class="column-names"
    >
      <div
        class="item"
      >
        id
      </div>
      <div
        class="item"
      >
        field1
      </div>
      <div
        class="item"
      >
        field2
      </div>
      <div
        class="item"
      >
        field3
      </div>
    </div>
    <div
      class="buttons"
    >
      Actions
    </div>
  </div>
</DocumentFragment>
`;
